--- src/latex.c	2023-09-08 09:01:12.688623000 +0800
+++ ../change/latex.c	2023-09-08 09:00:38.409355000 +0800
@@ -320,11 +320,20 @@
 
   case CMARK_NODE_CODE_BLOCK:
     CR();
-    LIT("\\begin{verbatim}");
+    LIT("\\begin{lstlisting}");
+    char langname[20];
+    strcpy(langname,node->as.code.info.data);
+    if(strcmp(langname,"shell-session"))strcpy(langname,"bash");
+    if( strcmp(langname,"sh")==0
+		    ||strcmp(langname,"bash")==0
+		    ||strcmp(langname,"csh")==0
+		    ||strcmp(langname,"ksh")==0
+		    ) {LIT("[language=");LIT(node->as.code.info.data);LIT("]");}
+
     CR();
     OUT(cmark_node_get_literal(node), false, LITERAL);
     CR();
-    LIT("\\end{verbatim}");
+    LIT("\\end{lstlisting}");
     BLANKLINE();
     break;
 
@@ -340,7 +349,7 @@
 
   case CMARK_NODE_THEMATIC_BREAK:
     BLANKLINE();
-    LIT("\\begin{center}\\rule{0.5\\linewidth}{\\linethickness}\\end{center}");
+    LIT("\\begin{center}\\rule{0.5\\textwidth}{1pt}\\end{center}");
     BLANKLINE();
     break;
 
@@ -438,10 +447,10 @@
 
   case CMARK_NODE_IMAGE:
     if (entering) {
-      LIT("\\protect\\includegraphics{");
+      LIT("\\protect\\resizebox{\\ifdim\\width>\\linewidth\\linewidth\\else\\width\\fi}{!}{\\includegraphics{");
       // requires \include{graphicx}
       OUT(cmark_node_get_url(node), false, URL);
-      LIT("}");
+      LIT("}}");
       return 0;
     }
     break;
